server/ladders.js

```diff
@@ -137,7 +137,7 @@ class Ladder extends LadderStore {
 		if (isRated && !Ladders.disabled) {
 			let userid = user.userid;
 			[valResult, rating] = await Promise.all([
-				TeamValidatorAsync(this.formatid).validateTeam(team, user.locked || user.namelocked),
+				TeamValidatorAsync(this.formatid).validateTeam(team, Boolean(user.locked || user.namelocked)),
```

Comment:
I'm not entirely sure why this leads to an error stating that the second parameter may possibly be "null". That should never be the case, given that if both user.locked or user.namelocked are `undefined|null`, the latter would just resolve to undefined and the default parameter of false would be used. Typescript weirdness?

Reply:
No, it's correct that if `user.locked` is falsy and `user.namelocked` is `null`, the end result would be `null`.

Anyway, PS uses the `!!` boolean cast; you used it correctly earlier.

```suggestion
				TeamValidatorAsync(this.formatid).validateTeam(team, !!(user.locked || user.namelocked)),
```

Reply:
woops, yeah that's better